Complete web and web2 merge

ivancuadra869 opened the pull request.

ivancuadra869 committed:
Add delete event-plan api call and fix buttons

ivancuadra869 committed:
Fix event-plan AvailabilityHeatMap colors

ivancuadra869 committed:
Fix FriendsPage button colors

ivancuadra869 committed:
Add scrolling to AvailabilityPage

ivancuadra869 committed:
Delete web folder

ivancuadra869 committed:
Remove button import in ConfirmEventModal

ivancuadra869 committed:
Rename web2 to web

navarrojandg commented on apps/web/package.json:
```diff
@@ -94,13 +94,13 @@
-    "clean": "rm -rf .turbo && rm -rf node_modules && rm -rf build"
+    "clean": "rm -rf .turbo && rm -rf node_modules && rm -rf build",
+    "build:tailwind": "tailwindcss build -i src/tailwind.css -o src/tailwind.output.css",
+    "prestart": "npm run build:tailwind"
```
Just get rid of this prestart script

dev: npm run build:tailwind && node scripts/start.js
build: npm run build:tailwind && node scripts/build.js

ivancuadra869 replied:
Okay, I'll try that, thank you sir!

navarrojandg replied:
No no just get rid of the prestart script!

You need the others!

dev: npm run build:tailwind && node scripts/start.js
build: npm run build:tailwind && node scripts/build.js

navarrojandg replied:
Instead of doing npm run prestart && ... you can just do npm run build:tailwind instead, you were like calling another npm script that calls another npm script when you can just call it directly.

navarrojandg replied:
Fixed

ivancuadra869 committed:
Remove dev and build  prestart scripts

navarrojandg committed:
Fix npm scripts in web

ivancuadra869 committed:
Remove tailwind.output.css

ivancuadra869 committed:
Change web2 to web in gitignore

ivancuadra869 committed:
Update autoprefixer

ivancuadra869 committed:
Fix build errors

jesc0bar committed:
downgrade autoprefixer and revert to working package-lock

ivancuadra869 commented:
Thank you Jorge for the build error fix.

navarrojandg committed:
Add tailwind.output.css to gitignore

navarrojandg commented on apps/web/src/components/EventPlanList/EventPlanList.tsx:
```diff
@@ -8,16 +6,14 @@ import {
 
-function _sortEventPlans(
+function sortEventPlans(
   eventPlans: (EventPlanInfo & { eventPlanId: EventPlanId } & {
```
This typing is kinda hard to understand -- my fault trying to make certain interfaces reusable, this was before reworking the functions api, back when we were only using firebase rules to control read / write. 

Will clean up the types and interfaces.

ivancuadra869 replied:
Okay awesome, are you going to making the changes that correspond to those types/interfaces?

navarrojandg replied:
Yes, just going thru rest of file changes to fix anything minor before doing so

ivancuadra869 commented:
Ok, can that be done on a separate branch?

On Sat, May 7, 2022, 10:33 AM Jan Navarro ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In apps/web/src/components/EventPlanList/EventPlanList.tsx
> <https://github.com/CSUN-COMP-490-Team-KBBQ/wya/pull/38#discussion_r867376032>
> :
>
> >  interface EventPlanListProps {
>    elementId: string;
>    eventPlans: (EventPlanInfo & { eventPlanId: EventPlanId } & {
>      isFinalized: EventPlanFinalizedFlag;
>    })[];
>  }
>
> -function _sortEventPlans(
> +function sortEventPlans(
>    eventPlans: (EventPlanInfo & { eventPlanId: EventPlanId } & {
>
> This typing is kinda hard to understand -- my fault trying to make certain
> interfaces reusable, this was before reworking the functions api, back when
> we were only using firebase rules to control read / write.
>
> Will clean up the types and interfaces.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/CSUN-COMP-490-Team-KBBQ/wya/pull/38#pullrequestreview-965350206>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ASZ432GILZNEAUY5AGRD4HLVI2SPFANCNFSM5VJTPMXA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

navarrojandg commented:
> Ok, can that be done on a separate branch?

Nah I'll commit the changes on here instead

navarrojandg commented on apps/web/src/components/RegisterForm/RegisterForm.tsx:
```diff
@@ -36,56 +32,189 @@ export default function RegisterForm(): JSX.Element {
+      setDisplayError('Passwords do not match!');
+    } else {
+      setDisplayError('');
+      await await api
```
await await is not correct.

Will fix this real quick.

ivancuadra869 replied:
Not sure what the heck happened there, thank you!

navarrojandg committed:
Clean up RegisterForm error handling

navarrojandg committed:
Add and handle http 422 errors from api

mtahery20 reviewed (dismissed):
works great! great job ivan!

navarrojandg commented on apps/web/src/components/RegisterForm/RegisterForm.tsx:
```diff
@@ -36,56 +32,208 @@ export default function RegisterForm(): JSX.Element {
 
     assert(recaptchaRef.current, 'ReCAPTCHA has not loaded');
-    const token = await recaptchaRef.current.executeAsync();
+    const token = recaptchaRef.current;
```
Token will always be null now. I know we don't really use the token anyway but just wanted to call out that captcha doesn't execute and we aren't capturing the token now.

navarrojandg commented on apps/web/src/index.tsx:
```diff
@@ -1,7 +1,8 @@
 import ReactDOM from 'react-dom';
 
-import './index.css';
+// import './index.css'; // to be removed?
```
To be removed?

ivancuadra869 replied:
not sure haha, that was in the web2 already

navarrojandg commented on apps/web/src/pages/EventPlanPage/EventPlanning.tsx:
```diff
@@ -1,10 +1,9 @@
 import React from 'react';
-import { Row, Col, Container, Modal, Button } from 'react-bootstrap';
+import { Modal } from 'react-bootstrap';
```
We are still using bootstrap modal? Or were we going to switch to tailwind

ivancuadra869 replied:
I thought so as well, but the creators of those pages kept some aspects I guess

navarrojandg commented on apps/web/src/pages/LandingPage/LandingPage.tsx:
```diff
@@ -1,51 +1,111 @@
+                <div className="flex items-center flex-grow flex-shrink-0">
+                  <div className="flex items-center justify-between w-full">
+                    {/* added for build to pass */}
+                    <a href="example.com">
```
What was causing the build to fail here?

ivancuadra869 replied:
Not sure, I didn't add this

ivancuadra869 commented:
Just so you know Jan, it looks like I added a lot in web, but in reality GitHub just thinks I added everything since I renamed web2 to web. It thinks all those changes are mine.

navarrojandg committed:
Add EventPlan type to web interfaces

navarrojandg reviewed (commented):
Did a first pass -- fixed anything that stood out to me. @ivancuadra869 Would need to deploy api again as made a change to the errors coming from `/users/create` route.

I like the way error messages are being shown to the user (saved in component state) but think they are going to persist even after the errors are fixed because we are not unsetting the error message state back to `''` or some falsey value. 

Also makes me feel like we are close to knowing the proper abstraction for showing positive / negative error messages to the user and can make it into a generic component so we can use it throughout the app.

navarrojandg committed:
Add confirmPassword to etlUserCreate

navarrojandg reviewed (approved):
There is a bootstrap modal still being used in one of the components, unsure if we want to keep that or not. But approving and we can fix that post merge.
